include/command_graph.h

```diff
@@ -1,6 +1,7 @@
 #pragma once
 
 #include <memory>
+#include <ostream>
```

Comment:
I don't like iostreams as an include dependency. Since this is for a debugging feature, would it be feasible to return `std::string` from the graph-printing facilities instead?

Reply:
Good point. `std::string` would probably work. We could also include `<iosfwd>` instead, either works for me.

Reply:
The graph printing functions now return `std::optional<std::string>`, as they will not print anything above a (configurable) threshold.

Reply:
Now you should be able to simply remove the `<ostream>` include.